test/commands/connect/diagnose.js

```diff
@@ -0,0 +1,151 @@
+/* globals describe beforeEach it */
+
+const cli = require('heroku-cli-util')
+const nock = require('nock')
+const expect = require('unexpected')
+const cmd = require('../../../commands/connect/diagnose')
+
+const password = 's3cr3t3'
+const headers = {
+  'content-type': 'application/json',
+  authorization: `Bearer ${password}`,
+  'heroku-client': 'cli'
+}
+
+describe('connect:diagnose', () => {
+  // prevent stdout/stderr from displaying
+  // redirects to cli.stdout/cli.stderr instead
+  beforeEach(() => cli.mockConsole())
+
+  it('runs validations with polling', () => {
+    let appName = 'fake-app'
+
+    let discoveryApi = nock('https://hc-central-qa.herokai.com/', { headers })
+      .get('/connections')
+      .query({ app: appName })
+      .reply(200, {
+        results: [
+          {
+            detail_url: 'https://hc-virginia-qa.herokai.com/connections/1234'
+          }
+        ]
+      })
+
+    const connectionData = {
+      id: 1234,
+      name: 'awesome-connection-1234',
+      db_key: 'DATABASE_URL',
+      state: 'IDLE',
+      schema_name: 'salesforce',
+      region_url: 'https://hc-virginia-qa.herokai.com/'
+    }
+
+    let connectionDetailApi = nock('https://hc-virginia-qa.herokai.com/', { headers })
+      .get('/connections/1234')
+      .query({ deep: true })
+      .reply(200, connectionData)
+
+    let connectionValidationApi = nock('https://hc-virginia-qa.herokai.com/', { headers })
+      .post('/api/v3/connections/1234/validations')
+      .reply(202, {
+        result_url: 'https://hc-virginia-qa.herokai.com/api/v3/connections/1234/validations/456'
+      })
+      .get('/api/v3/connections/1234/validations/456')
+      .twice()
+      .reply(202)
+      .get('/api/v3/connections/1234/validations/456')
+      .reply(200, {
+        // data: {
+        passes: [
+          {
+            rule_id: 'NUM_MAPPINGS_WITHIN_LIMIT',
+            display_name: 'Number of Mappings',
+            status: 'PASSED',
+            message: 'Connection has no more than 100 mappings.',
+            doc_url:
+                'https://devcenter.heroku.com/articles/heroku-connect-diagnose#check-number-of-mappings'
+          },
+          {
```

Comment:
It doesn't look like we're asserting all of these, so can we get rid of the unnecessary ones?

Reply:
good call, will do!